src/battle_ai_util.c

```diff
@@ -3249,82 +3285,55 @@ bool32 IsRecycleEncouragedItem(u32 item)
     return FALSE;
 }
 
-// score increases
-#define STAT_UP_2_STAGE     8
-#define STAT_UP_STAGE       10
 void IncreaseStatUpScore(u32 battlerAtk, u32 battlerDef, u32 statId, s32 *score)
 {
+    u32 noOfHitsToFaint = NoOfHitsForTargetToFaintAI(battlerDef, battlerAtk);
+    u32 aiIsFaster = GetWhichBattlerFaster(battlerAtk, battlerDef, TRUE) == AI_IS_FASTER;
+
     if (AI_DATA->abilities[battlerAtk] == ABILITY_CONTRARY)
         return;
 
-    if (AI_DATA->hpPercents[battlerAtk] < 80 && AI_RandLessThan(128))
+    // Don't increase stat if AI is at +4
+    if (gBattleMons[battlerAtk].statStages[statId] >= MAX_STAT_STAGE - 2)
         return;
 
-    if ((AI_THINKING_STRUCT->aiFlags[battlerAtk] & AI_FLAG_TRY_TO_FAINT) && CanAIFaintTarget(battlerAtk, battlerDef, 0))
-        return; // Damaging moves would get a score boost from AI_TryToFaint or PreferStrongestMove so we don't consider them here
+    // Don't increase stat if AI has less then 70% HP and number of hits isn't known
+    if (AI_DATA->hpPercents[battlerAtk] < 70 && noOfHitsToFaint == 0)
+        return;
 
     switch (statId)
     {
     case STAT_ATK:
-        if (HasMoveWithCategory(battlerAtk, BATTLE_CATEGORY_PHYSICAL) && AI_DATA->hpPercents[battlerAtk] > 40)
-        {
-            if (gBattleMons[battlerAtk].statStages[STAT_ATK] < STAT_UP_2_STAGE)
-                ADJUST_SCORE_PTR(2);
-            else if (gBattleMons[battlerAtk].statStages[STAT_ATK] < STAT_UP_STAGE)
-                ADJUST_SCORE_PTR(1);
-        }
-        if (HasMoveEffect(battlerAtk, EFFECT_FOUL_PLAY))
-            ADJUST_SCORE_PTR(1);
+        if ((HasMoveWithCategory(battlerAtk, BATTLE_CATEGORY_PHYSICAL))
+        && ((noOfHitsToFaint >= 2 && aiIsFaster) || (noOfHitsToFaint >= 3 && !aiIsFaster) || noOfHitsToFaint == 0))
+            ADJUST_SCORE_PTR(4);
```

Comment:
Possibly slightly out of scope, but I think at some point we should start making macros for score points, because e.g. all the stats here use 4 now, but given the consistency it could perhaps be clearer if it weren't a magic number but rather a macro somewhere

Reply:
Not convinced on macros but we can think about up defines for score increases.

Reply:
I meant defines rather than macros (especially since `ADJUST_SCORE_PTR` is already a macro), my bad

Reply:
```
# define WEAK_EFFECT 1
# define DECENT_EFFECT 2
# define GOOD_EFFECT 4
# define BEST_EFFECT 6
```
Open to better name suggestions. I also think we don't need 3 and 5 with the way I set up the score increases.

Reply:
Considering it out of scope is totally fair to me, no worries! I'll go ahead and merge this one then